### ISSUE TITLE
Transitions toAncestor or toDescendant

### ISSUE THREAD

**Issue by cjenaro** [roxiness/routify#196]
The following I believe is a bug, but not sure of it.

**bug description**
When navigating from/to pages/index.svelte to pages/anything.svelte, the toAncestor and toDescendant are both false, given that the filepath is pages/something.svelte, I assume, but being so that the pages/index.svelte corresponds to the path "/" and anything.svelte to "/anything" I believe anything should be a descendant to index.

Here's what I did to mock this behaviour:

```js
<script>
  import { scale, fly } from 'svelte/transition'
  import { BaseTransition } from "@sveltech/routify/decorators"
  export let scoped
  const { width } = scoped
  const configs = [
    {
      condition: ({ routes }) => routes[0] === routes[1],
      transition: () => {},
    },
    {
      condition: c => c.toDescendant || c.routes[1].path.includes('index'),
      transition: fly,
      inParams: { x: $width, duration: 500, opacity: 1 },
      outParams: { x: -$width, duration: 500, opacity: 1 },
    },
    {
      condition: c => c.toAncestor || c.routes[0].path.includes('index'),
      transition: fly,
      inParams: { x: -$width, duration: 500, opacity: 1 },
      outParams: { x: $width, duration: 500, opacity: 1 },
    },
    {
      // No matching config. We don't want a transition
      condition: (c) => {
        console.log(c)
        return true
      },
      transition: () => {},
    },
  ]
</script>

<BaseTransition {configs}>
  <slot />
</BaseTransition>
```

This code is not bulletproof, of course, it lives [here](https://github.com/jenaro94/routify-pwa-starter), it's a routify pwa template I'm building.

**Comment by cjenaro** [comment]
I can see this method on Base Transition

```js
  function isAncestor(descendant, ancestor) {
    if(descendant.parent === ancestor.parent) return false
    const { shortPath } = descendant.parent
    const { shortPath: shortPath2 } = ancestor.parent
    
    return ancestor.isIndex && shortPath !== shortPath2 && shortPath.startsWith(shortPath2)
  }
```

and aparently there is a check to see if ancestor is index? For some reason it is not working for me at the template I linked.

**Comment by Wolfr** [comment]
If you click from one page to a deeper page, this is what `condition` returns

```{toAncestor: false, toDescendant: false, toHigherIndex: false, toLowerIndex: false, routes: Array(2), …}```

I would expect `toDescendant` to be true in this case. I can confirm the bug.

@jenaro94 it would probably help to link to the full repo.

**Comment by jakobrosenberg** [comment]
Are you able to get `toAncestor` or `toDescendant`at all?

**Comment by Wolfr** [comment]
I deployed a version of a demo that shows a reduced test case.

https://routify-slide-transitions-demo.now.sh/

Repo: https://github.com/Wolfr/routify-slide-transition-demo

BTW this demo has bugs, obviously :) I have gotten it to work locally. But on the deploy the transition is not working at all.

**Comment by Wolfr** [comment]
A bit confused... 

If I log `descendant` from this code, I get a page object, but it doesn't have any `parent` child. Is it suppoed to have one?

```
  function isAncestor(descendant, ancestor) {
    console.log(descendant);
    if(descendant.parent === ancestor.parent) return false
    const { shortPath } = descendant.parent
    const { shortPath: shortPath2 } = ancestor.parent
    
    return ancestor.isIndex && shortPath !== shortPath2 && shortPath.startsWith(shortPath2)
  }
```

![image](https://user-images.githubusercontent.com/12690/83353303-f1a4a880-a351-11ea-8dc7-1316a349f815.png)

**Comment by Wolfr** [comment]
If I click open API I can see it is exposed on the ClientApi?

![image](https://user-images.githubusercontent.com/12690/83353386-8ad3bf00-a352-11ea-9904-cc3e5ca82425.png)

This is maybe a bit too complex for me 😅

**Comment by cjenaro** [comment]
I'm sorry I linked it but it isn't very apparent haha, the repo I used is https://github.com/jenaro94/routify-pwa-starter

**Comment by cjenaro** [comment]
I also logged ancestor here, and got:

![Screenshot from 2020-05-31 11-18-12](https://user-images.githubusercontent.com/15108197/83354703-be9ded00-a330-11ea-94ce-5ae928a7e7f1.png)

So the file clearly is index.svelte, but the isIndex property is still false.

**Comment by cjenaro** [comment]
Playing with the isAncestor function: 

```js
  function isAncestor(descendant, ancestor) {
      /* if(descendant.parent === ancestor.parent) return false */
      const { shortPath } = descendant.parent
      const { shortPath: shortPath2 } = ancestor.parent
      console.log(descendant.filepath.split('/'), ancestor.filepath.split('/'))
      const descendantLength = descendant.filepath.split('/').length
      const ancestorLength = ancestor.filepath.split('/').length
      return descendantLength > ancestorLength || descendantLength === ancestorLength && ancestor.isIndex
      /* return ancestor.isIndex && shortPath !== shortPath2 && shortPath.startsWith(shortPath2) */
    }
```

It seems to be working, or at least it is covering the case where I had the issue.

**Comment by cjenaro** [comment]
Had to add a check to see if filepath existed on the build version, you can check it out at https://routify-pwa-starter.surge.sh

**Comment by jakobrosenberg** [comment]
This was fixed in 2.0.

Closing till 2.0 is released.
